### PR TITLE
Upgrade nginx-hugo-theme version to support newer Hugo releases

### PR DESCRIPTION
**Pull request by jputrino**
### Proposed changes

This PR updates the nginx-hugo-theme module to v0.34.0, which adds support for Hugo v0.115.3 (and subsequent newer releases). This upgrade is necessary to address security vulnerabilities in the earlier pinned version of Hugo.

- The docs/README has been updated to match that used across the NGINX projects where we develop docs.
- The docs/makefile and netlify configs have been updated to include the deploy-preview make command, which is needed to build docs for the deploy-preview and branch contexts in Netlify. These settings are also now in parity with the configurations used across the NGINX documentation projects.

Related PR: https://github.com/nginxinc/nginx-hugo-theme/pull/16

Once this PR is merged, the environment variables that are stored in Netlify for the nginx-hugo-theme version and hugo version will be updated accordingly.

### Checklist

Before creating a PR, run through this checklist and mark each as complete.

- [x] I have read the [CONTRIBUTING](https://github.com/nginxinc/kubernetes-ingress/blob/main/CONTRIBUTING.md) doc
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that all unit tests pass after adding my changes
- [x] I have updated necessary documentation
- [x] I have rebased my branch onto main
- [ ] I will ensure my PR is targeting the main branch and pulling from my branch from my own fork

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -1,24 +1,136 @@
...
+make hugo-mod       -   cleans the Hugo module cache and fetches the latest version of the theme module
+make docs-drafts    -   runs the local hugo server and includes all docs marked with `draft: true`
+```
+
+## Linting
+
+- To run the markdownlint check, run the following command from the docs directory:
+
+    ```bash
+    markdownlint -c docs/mdlint_conf.json content
```
Comment (lucacome) [inline docs/README.md:35]: do we need to add this file `docs/mdlint_conf.json` ?

Comment (lucacome) [reply docs/README.md:35]: we just recently added `.markdownlint-cli2.yaml` to the root of the project for all other markdown files, but I'm not sure if we want to apply the same rules here for the docs

Comment (jputrino) [reply docs/README.md:35]: I'll take a look, thanks for catching that!

Comment (jputrino) [reply docs/README.md]: @lucacome I'm taking this out of the README and we will address it in a separate PR. It's a much larger can of worms than I care to bring into the context of this PR.

Comment (jputrino) [reply docs/README.md]: We don't want to use the main markdownlint config file for the docs, but we need to standardize on the rules across all of the repos before going further.

**Review by lucacome — APPROVED** [review]

**Pull request merged**